lib/Driver/DarwinToolChains.cpp

```diff
@@ -588,8 +588,66 @@ bool toolchains::Darwin::shouldStoreInvocationInDebugInfo() const {
 void 
 toolchains::Darwin::validateArguments(DiagnosticEngine &diags,
                                       const llvm::opt::ArgList &args) const {
+  // Validating bridging header files.
+  validateBridgingHeaderArgs(diags, args);
+
+  // Validating apple platforms deployment targets.
+  validateDeploymentTarget(diags, args);
+    
   // Validating darwin unsupported -static-stdlib argument.
   if (args.hasArg(options::OPT_static_stdlib)) {
     diags.diagnose(SourceLoc(), diag::error_darwin_static_stdlib_not_supported);
   }
 }
+
+void
+toolchains::Darwin::validateDeploymentTarget(DiagnosticEngine &diags,
+                                             const llvm::opt::ArgList &args) const {
+  const Arg *A = args.getLastArg(options::OPT_target);
+  if (!A)
+    return;
+  
+  // Check minimum supported OS versions.
+  llvm::Triple triple(llvm::Triple::normalize(A->getValue()));
+  if (triple.isMacOSX()) {
+    if (triple.isMacOSXVersionLT(10, 9))
+      diags.diagnose(SourceLoc(), diag::error_os_minimum_deployment,
+                     "OS X 10.9");
+  } else if (triple.isiOS()) {
+    if (triple.isTvOS()) {
+      if (triple.isOSVersionLT(9, 0)) {
+        diags.diagnose(SourceLoc(), diag::error_os_minimum_deployment,
+                       "tvOS 9.0");
+        return;
+      }
+    }
+    if (triple.isOSVersionLT(7))
+      diags.diagnose(SourceLoc(), diag::error_os_minimum_deployment,
+                     "iOS 7");
+    if (triple.isArch32Bit() && !triple.isOSVersionLT(11)) {
+      diags.diagnose(SourceLoc(), diag::error_ios_maximum_deployment_32,
+                     triple.getOSMajorVersion());
+    }
+  } else if (triple.isWatchOS()) {
+    if (triple.isOSVersionLT(2, 0)) {
+      diags.diagnose(SourceLoc(), diag::error_os_minimum_deployment,
+                     "watchOS 2.0");
+      return;
+    }
+  }
+}
+
+void
+toolchains::Darwin::validateBridgingHeaderArgs(DiagnosticEngine &diags,
```

Comment:
We don't have a good story for these flags on non-Apple platforms, but if I'm remembering correctly they're currently accepted and therefore these conditions should probably still be enforced everywhere.

Reply:
Sure, just reverted :)